drivers/entropy/entropy_stm32.c

```diff
@@ -114,19 +119,49 @@ static int entropy_stm32_got_error(RNG_TypeDef *rng)
 	return 0;
 }
 
+#if defined(STM32_CONDRST_SUPPORT)
+static int recover_seed_error(RNG_TypeDef *rng)
+{
+	uint32_t count_timeout = 0;
+
+	LL_RNG_EnableCondReset(rng);
+	LL_RNG_DisableCondReset(rng);
+	/* When reset process is done cond reset bit is read 0
+	 * This typically takes: 2 AHB clock cycles + 2 RNG clock cycles.
+	 */
+
+	while (LL_RNG_IsEnabledCondReset(rng) ||
+		LL_RNG_IsActiveFlag_SECS(rng)) {
+		count_timeout++;
+		if (count_timeout == 10) {
+			return -ETIMEDOUT;
+		}
+	}
+
+	return 0;
+}
+#endif /* STM32_CONDRST_SUPPORT */
+
 static int random_byte_get(void)
 {
 	int retval = -EAGAIN;
 	unsigned int key;
+	RNG_TypeDef *rng = entropy_stm32_rng_data.rng;
 
 	key = irq_lock();
 
-	if ((LL_RNG_IsActiveFlag_DRDY(entropy_stm32_rng_data.rng) == 1)) {
-		if (entropy_stm32_got_error(entropy_stm32_rng_data.rng)) {
+#if defined(STM32_CONDRST_SUPPORT)
+	if (LL_RNG_IsActiveFlag_SECS(rng) && (recover_seed_error(rng) < 0)) {
+		irq_unlock(key);
+		return -EIO;
+	}
+#endif /* STM32_CONDRST_SUPPORT */
+
+	if ((LL_RNG_IsActiveFlag_DRDY(rng) == 1)) {
+		if (entropy_stm32_got_error(rng)) {
 			retval = -EIO;
 		} else {
-			retval = LL_RNG_ReadRandData32(
-						    entropy_stm32_rng_data.rng);
+			retval = LL_RNG_ReadRandData32(rng);
```

Comment:
According to Reference Manual RM0438 (stm32l5):

> Note: When data is not ready (DRDY=”0”) RNG_DR returns zero.
> **It is recommended to always verify that RNG_DR is different from zero**. Because when it is
> the case a seed error occurred between RNG_SR polling and RND_DR output reading (rare
> event).

So it is necessary to check **retval** and if zero,  call recover_seed_error() or return an error.

Reply:
Thanks. Checking several RMs, this seems to apply to most series, some series don't call it out, but don't list any different possibly invalid reg-value. So I will apply this for all series and push the changes together with the requests above.